src/xrpld/app/tx/detail/VaultWithdraw.cpp

```diff
@@ -307,11 +314,57 @@ VaultWithdraw::doApply()
         // else quietly ignore, account balance is not zero
     }
 
-    auto const dstAcct = [&]() -> AccountID {
-        if (ctx_.tx.isFieldPresent(sfDestination))
-            return ctx_.tx.getAccountID(sfDestination);
-        return account_;
-    }();
+    auto const dstAcct = ctx_.tx[~sfDestination].value_or(account_);
+
+    if (!vaultAsset.native() && dstAcct != vaultAsset.getIssuer() &&
+        dstAcct == account_)
+    {
+        if (auto const ter = std::visit(
+                [&]<ValidIssueType TIss>(TIss const& issue) -> TER {
+                    if constexpr (std::is_same_v<TIss, Issue>)
+                    {
+                        std::shared_ptr<SLE> dummy = {};
+                        if (auto const sleLine =
+                                view().read(keylet::line(dstAcct, issue));
+                            sleLine == nullptr)
+                        {
+                            auto const sleAcct =
+                                view().peek(keylet::account(account_));
+                            if (!sleAcct)
+                                return tecINTERNAL;  // LCOV_EXCL_LINE
+
+                            // Can the account cover the trust line reserve ?
+                            std::uint32_t const ownerCount =
+                                sleAcct->at(sfOwnerCount);
+                            if (mPriorBalance <
+                                view().fees().accountReserve(ownerCount + 1))
+                                return tecNO_LINE_INSUF_RESERVE;
+                        }
+
+                        // If the above succeeded then the required trust line
+                        // will be created indirectly by `accountSend` below.
+                    }
+                    else
+                    {
+                        if (auto const sleMPT = view().read(
```

Comment:
This is more efficient:

     if (!view().exists(keylet::mptoken(issue.getMptID(), account_)))